span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -569,13 +569,27 @@ private static URL getNormalizedUrl(String url) throws MalformedURLException {
     return new URL(new URL(RELATIVE_URL_CONTEXT), url);
   }
 
+  /**
+   * If the requestBuilder already has absolute url, do nothing
+   * if not, try building url based on otel attributes and overwrite
+   */
   private void maybeSetHttpUrlForOtelFormat(
       Request.Builder requestBuilder,
       final Map<String, AttributeValue> attributeValueMap) {
-    if (requestBuilder.hasUrl()) {
+    if (requestBuilder.hasUrl() && isAbsoluteUrl(requestBuilder.getUrl())) {
       return;
     }
     Optional<String> url = HttpSemanticConventionUtils.getHttpUrlForOTelFormat(attributeValueMap);
     url.ifPresent(requestBuilder::setUrl);
   }
+
+  static boolean isAbsoluteUrl(String urlStr) {
+    try {
+      URL url = getNormalizedUrl(urlStr);
+      return url.toString().equals(urlStr);
+    } catch (MalformedURLException e) {
```

Comment:
Can we add unit tests for these negative scenarios?(If possible)

Reply:
The java implementation for URL conversion seems to be able to parse all the strings! I will update the test, when I find one.